akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -370,6 +372,7 @@ private[stream] final class VirtualProcessor[T] extends Processor[T, T] {
           try {
             subscriptionStatus.set(s)
             tryOnSubscribe(s, sub)
+            if (isCancelled) throw canNotSubscribeTheSameSubscriberMultipleTimesException
```

Comment:
This violates Reactive Streams spec rule 1:9: https://github.com/reactive-streams/reactive-streams-jvm/blob/v1.0.0/README.md#specification